src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -85,10 +128,20 @@ service::service(
 ss::future<> service::start() {
     static std::vector<model::broker_endpoint> not_advertised{};
     _server.routes(get_schema_registry_routes(_gate));
-    return _server.start(
+    co_await _server.start(
       _config.schema_registry_api(),
       _config.schema_registry_api_tls(),
       not_advertised);
+    (void)ss::with_gate(_gate, [this]() -> ss::future<> {
+        try {
+            co_await create_internal_topic();
```

Comment:
should this also be kicked off on the error path from interacting with the schema topic if it happens to not exist (e.g. if this failed on startup to create the internal topic?)

Reply:
> should this also be kicked off on the error path from interacting with the schema topic if it happens to not exist (e.g. if this failed on startup to create the internal topic?)

This is at startup. Do I misunderstand?

Reply:
> > should this also be kicked off on the error path from interacting with the schema topic if it happens to not exist (e.g. if this failed on startup to create the internal topic?)
> 
> This is at startup. Do I misunderstand?

Oh, you mean in addition to at startup. Makes sense.

Reply:
See `one_shot`